ethereumj-core/src/main/java/org/ethereum/db/RepositoryImpl.java

```diff
@@ -147,7 +147,7 @@ public synchronized void saveCode(byte[] addr, byte[] code) {
     @Override
     public byte[] getCodeHash(byte[] addr) {
         AccountState accountState = getAccountState(addr);
-        return accountState != null ? accountState.getCodeHash() : HashUtil.EMPTY_DATA_HASH;
+        return accountState != null ? accountState.getCodeHash() : null;
```

Comment:
Are you sure that this change won't break anything else than `getCode`?

Reply:
@mkalinin checked it with extra attention, it shouldn't break anything for us